### DISCUSSION TITLE
Sever-Side Session

### DISCUSSION THREAD

**Discussion by rmartin93** [vercel/next.js#62457] [Help]
### Summary

Does anybody know of a good way to create Server-Side session variables in Next 14? 

I can't find any documentation about it. The cookie thing looks okay, but I don't like that I can't use it in my root layout component. 

I'd like to be able to store more than just user data in my sessions but maybe next just isn't built for that. 

If anybody knows of any good articles on this topic I'd appreciate a link. 

I've done a lot of Googling and I can't find any good info on this. Clerk seems to have figured out how to do it, but I want to use my database for creating a bunch of session data. Clerk has a publicMetaData thing but the session token has a max size of 4kb so that's no good. 

I'm just kind of rambling now, but it seems crazy that there's not a straight-forward way to create session data right? Context works for straight up react apps, but in the Next world my server components can't access context so I'm just kinda screwed. 

Anyway, if anybody knows more about this stuff than I do feel free to chime in.

**Comment by Enkratia** [comment]
You mean `next-auth` features to get session/token in server component?

If yes, `next-auth` have 2 methods:
- getToken()
- getServerSession()

https://next-auth.js.org/tutorials/securing-pages-and-api-routes#using-gettoken
